### ISSUE TITLE
Gradient accumulation and OOM during training

### ISSUE THREAD

**Issue by beaglew** [guandeh17/Self-Forcing#15]
Hi, thanks for the great work!
Is there anyone know how to implement gradient accumulation?
Besides, I only have A100 GPUs and find the default training setting cause OOM. Is there any solution? Currently, I just reduce the training data size.

**Comment by beaglew** [comment]
> Hi, thanks for the great work! Is there anyone know how to implement gradient accumulation? Besides, I only have A100 GPUs and find the default training setting cause OOM. Is there any solution? Currently, I just reduce the training data size.

I find OOM when I use 2 GPUs, but 4 GPUs not.

**Comment by George0726** [comment]
> > Hi, thanks for the great work! Is there anyone know how to implement gradient accumulation? Besides, I only have A100 GPUs and find the default training setting cause OOM. Is there any solution? Currently, I just reduce the training data size.
> 
> I find OOM when I use 2 GPUs, but 4 GPUs not.

Hi could you mind sharing your configs? I use 4A100 GPUs, but I still encounter OOM.
Here I attached my config and running code


############################
GPUS_PER_NODE=4
DISTRIBUTED_ARGS="
    --rdzv-backend=c10d \
    --rdzv-endpoint=localhost:6009 \
    --nnodes=1 --nproc_per_node=4
"
export PYTHONPATH="$MAGI_ROOT:$PYTHONPATH"
CUDA_VISIBLE_DEVICES=0,1,2,4 torchrun $DISTRIBUTED_ARGS train.py --config_path configs/self_forcing_dmd.yaml \
  --logdir logs/self_forcing_dmd_node \
  --disable-wandb 



############################
generator_ckpt: checkpoints/ode_init.pt
generator_fsdp_wrap_strategy: size
real_score_fsdp_wrap_strategy: size
fake_score_fsdp_wrap_strategy: size
real_name: Wan2.1-T2V-14B
fake_name: Wan2.1-T2V-14B
text_encoder_fsdp_wrap_strategy: size
denoising_step_list:
- 1000
- 750
- 500
- 250
warp_denoising_step: true # need to remove - 0 in denoising_step_list if warp_denoising_step is true
ts_schedule: false
num_train_timestep: 1000
timestep_shift: 5.0
guidance_scale: 3.0
denoising_loss_type: flow
mixed_precision: true
seed: 0
wandb_host: WANDB_HOST
wandb_key: WANDB_KEY
wandb_entity: WANDB_ENTITY
wandb_project: WANDB_PROJECT
sharding_strategy: hybrid_full
lr: 2.0e-06
lr_critic: 4.0e-07
beta1: 0.0
beta2: 0.999
beta1_critic: 0.0
beta2_critic: 0.999
data_path: prompts/vidprom_filtered_extended.txt
batch_size: 1
ema_weight: 0.99
ema_start_step: 200
total_batch_size: 1
log_iters: 50
negative_prompt: '色调艳丽，过曝，静态，细节模糊不清，字幕，风格，作品，画作，画面，静止，整体发灰，最差质量，低质量，JPEG压缩残留，丑陋的，残缺的，多余的手指，画得不好的手部，画得不好的脸部，畸形的，毁容的，形态畸形的肢体，手指融合，静止不动的画面，杂乱的背景，三条腿，背景人很多，倒着走'
dfake_gen_update_ratio: 5
image_or_video_shape:
- 1
- 21
- 16
- 60
- 104
distribution_loss: dmd
trainer: score_distillation
gradient_checkpointing: true
num_frame_per_block: 3
load_raw_video: false
model_kwargs:
  timestep_shift: 5.0

**Comment by LeeKeyu** [comment]
@beaglew Hi, may I ask if you used 40GB or 80GB A100 to reproduce? Also, I would also like to know how to modify to train on 40GB GPUs. Thanks a lot in advance!

**Comment by beaglew** [comment]
> [@beaglew](https://github.com/beaglew) Hi, may I ask if you used 40GB or 80GB A100 to reproduce? Also, I would also like to know how to modify to train on 40GB GPUs. Thanks a lot in advance!

Hey, I use 80G A100.

**Comment by haodong2000** [comment]
Hi @beaglew, may I ask how many GPU memory is cost when using 4 80G A100s? Thanks!

**Comment by beaglew** [comment]
> Hi [@beaglew](https://github.com/beaglew), may I ask how many GPU memory is cost when using 4 80G A100s? Thanks!

maybe 60~70GB
